src/Translator/TranslationProxy.php

```diff
@@ -18,25 +18,41 @@
  */
 class TranslationProxy
 {
+    /**
+     * @var string
+     */
     protected $locale;
+    /**
+     * @var object
+     */
     protected $translatable;
+    /**
+     * @var string[]
+     */
     protected $properties = [];
+    /**
+     * @var string
+     *
+     * @phpstan-var class-string<TranslationInterface>
+     */
     protected $class;
     /**
-     * @var Collection|TranslationInterface[]
+     * @var Collection<TranslationInterface>
```

Comment:
For `Collection` I think you must specify also the key type: https://github.com/doctrine/collections/blob/42a111afab917a328fdc38a35101c089ce52b580/lib/Doctrine/Common/Collections/Collection.php#L27-L32.

Since here I think it is only used `$this->coll->add($translation);` I would use `int` as key type.